cassiopeia/simulator/LineageTracingDataSimulator.py

```diff
@@ -0,0 +1,25 @@
+"""
+This file stores an abstract subclass of DataSimulator, the
+LineageTracingDataSimulator. A LineageTracingDataSimulator overlays lineage
+tracing data onto a CassiopeiaTree, i.e. it sets the character states of a
+CassiopeiaTree (in particular, the character matrix).
+"""
+import abc
+from cassiopeia.data import CassiopeiaTree
+from cassiopeia.simulator.DataSimulator import DataSimulator
+
+
+class LineageTracingDataSimulator(DataSimulator):
+    """
+    LineageTracingDataSimulator is an abstract class that all lineage tracing
+    data simulators derive from.
```

Comment:
Can you expand a bit more on what exactly "lineage tracing data" is? What should this populate in the `CassiopeiaTree`? When should users use this?

Reply:
I mentioned that the character states of the tree are set as well as the character matrix in the doscstring of the `overlay_data` method. Should these details be duplicated here in the class' docstring? A general question I have is what we want to say in the class' dosctring vs the method's docstring vs the module's dosctring.